app/forms/ReferenceNumberFormProvider.scala

```diff
@@ -16,17 +16,20 @@
 
 package forms
 
-import javax.inject.Inject
-
 import forms.mappings.Mappings
 import play.api.data.Form
 
+import javax.inject.Inject
 class ReferenceNumberFormProvider @Inject() extends Mappings {
 
   def apply(): Form[String] =
     Form(
-      "value" -> text("referenceNumber.error.required")
-        .verifying(minLength(2, "referenceNumber.error.length"))
-        .verifying(maxLength(64, "referenceNumber.error.length"))
+      "value" -> text("referenceNumber.error.required").transform[String](_.replaceAll("\\s", ""), identity)
+        .verifying(firstError(
+          regexp(Validation.referenceNumberPattern,"referenceNumber.error.invalid.chars"),
+          startsWith("referenceNumber.error.prefix"),
+          minLength(23, "referenceNumber.error.length"),
+          maxLength(23, "referenceNumber.error.length")
```

Comment:
Since we have the same error message for both of these could we use inRange instead of min and max checks?

Reply:
Cool didn't know that existed :)

Reply:
The existing inRange works only for Int